packages/vue-apollo-composable/src/useApolloClient.ts

```diff
@@ -23,31 +23,35 @@ function resolveDefaultClient<T>(providedApolloClients: ClientDict<T>, providedA
 }
 
 function resolveClientWithId<T>(providedApolloClients: ClientDict<T>, clientId: ClientId): ApolloClient<T> {
-  if (!providedApolloClients) {
-    throw new Error(`No apolloClients injection found, tried to resolve '${clientId}' clientId`)
-  }
   const resolvedClient = providedApolloClients[clientId]
   if (!resolvedClient) {
     throw new Error(`Apollo Client with id ${clientId} not found`)
   }
   return resolvedClient
 }
 
+function assertProvidedApolloClients<T>(providedApolloClients: ClientDict<T>, clientId: ClientId) {
+  if (!providedApolloClients) {
+    throw new Error(`No apolloClients injection found, tried to resolve '${clientId}' clientId`)
+  }
+}
+
 export function useApolloClient<TCacheShape = any>(clientId?: ClientId): UseApolloClientReturn<TCacheShape> {
-  const providedApolloClients: ClientDict<TCacheShape> = inject(ApolloClients, null)
-  const providedApolloClient: ApolloClient<TCacheShape> = inject(DefaultApolloClient, null)
+  const providedApolloClients = inject<ClientDict<TCacheShape>>(ApolloClients, null)
+  const providedApolloClient = inject<ApolloClient<TCacheShape>>(DefaultApolloClient, null)
 
-  function resolveClient() {
+  function resolveClient(clientId?: ClientId) {
     if (clientId) {
-      resolveClientWithId(providedApolloClients, clientId)
+      assertProvidedApolloClients(providedApolloClients, clientId)
+      return resolveClientWithId(providedApolloClients, clientId)
```

Comment:
before just exec resolveClientWithId. not return.
And then resolveClient required optional clientId params for follow UseApolloClientReturn interface.

Reply:
https://github.com/vuejs/vue-apollo/commit/72d7402eb29893a7e63fc675a8057a8f5a4a825f